### DISCUSSION TITLE
I can't get some midi actions to work

### DISCUSSION THREAD

**Discussion by remekdc** [hydrogen-music/hydrogen#1502] [Q&A]
Hey so first of all: thanks to everyone who has build this program. I really love it!

I'm trying to use it on a windows machine. Hydrogen 1.1.1.
I'm using my phone as a midi device via USB using the app Pocket Midi.

The computer picks it up. Hydrogen picks up the inputs (and works for PLAY/STOP, PLAYLIST_PREV_SONG etc). However, there are a few actions that hydrogen doesn't seem to want to make work for me

The two specifically I am looking at are

TAP TEMPO
and
SELECT_NEXT_PATTERN

Tap tempo just doesn't seem to do anything...

With regards to select next pattern: do I understand correctly that that if I select the next pattern and I am on Pattern 1 it would go to Pattern 2 (which is the next pattern  on the list)? At the moment. Whatever I start with (let's say pattern 6 to start) it'll go straight Pattern 1. I'm using "Pattern" Mode as opposed to "Song" mode

Is there something I've done wrong?
Thanks for your time

**Comment by theGreatWhiteShark** [comment]
Hey @remekdc ,

> Tap tempo just doesn't seem to do anything...

This one is used to set the tempo and has to be triggered multiple times in a row. Hydrogen will interpret the incoming TAP_TEMPO messages as quarters of a beat and calculates the average tempo from the last 10. For this to work the messages have to be received in a row with not too much time passing between them and a certain amount of them (3-4?) are required at a minimum.

> With regards to select next pattern: do I understand correctly that that if I select the next pattern and I am on Pattern 1 it would go to Pattern 2 (which is the next pattern on the list)? At the moment. Whatever I start with (let's say pattern 6 to start) it'll go straight Pattern 1. I'm using "Pattern" Mode as opposed to "Song" mode

Almost. `SELECT_NEXT_PATTERN` selects pattern `n` using the pattern number (order of patterns in the Song editor). `SELECT_NEXT_PATTERN_RELATIVE` switches `n` patterns ahead or back depending on the current pattern number.
